raphtory/src/python/utils/mod.rs

```diff
@@ -71,11 +75,45 @@ impl<'source> FromPyObject<'source> for PyTime {
         if let Ok(parsed_datetime) = time.extract::<NaiveDateTime>() {
             return Ok(PyTime::new(parsed_datetime.try_into_time()?));
         }
-        let message = format!("time '{time}' must be a str, dt or an integer");
+        if let Ok(py_datetime) = time.extract::<&PyDateTime>() {
+            let naive_date = NaiveDate::from_ymd_opt(
+                py_datetime.get_year(),
+                py_datetime.get_month() as u32,
+                py_datetime.get_day() as u32,
+            );
+            let naive_time = NaiveTime::from_hms_micro_opt(
+                py_datetime.get_hour() as u32,
+                py_datetime.get_minute() as u32,
+                py_datetime.get_second() as u32,
+                py_datetime.get_microsecond(),
+            );
+
+            if let (Some(naive_date), Some(naive_time)) = (naive_date, naive_time) {
+                let naive_datetime = NaiveDateTime::new(naive_date, naive_time);
+                // Handle timezone
+                let tz_offset = if let Some(tz) = py_datetime.get_tzinfo() {
+                    let offset = tz
+                        .call_method1("utcoffset", (py_datetime,))?
+                        .extract::<&PyDelta>()?;
+                    let offset_seconds =
+                        offset.call_method0("total_seconds")?.extract::<f64>()? as i32;
+                    FixedOffset::east_opt(offset_seconds)
+                        .ok_or(PyTypeError::new_err("Could not pass timezone"))
+                } else {
+                    FixedOffset::east_opt(0).ok_or(PyTypeError::new_err("Could not pass timezone"))
+                    // No timezone information, assume UTC
+                }?;
+
+                let utc_datetime = tz_offset.from_local_datetime(&naive_datetime).unwrap();
+                let timestamp = utc_datetime.timestamp() * 1000;
+                return Ok(PyTime::new(timestamp.try_into_time()?));
+            }
+        }
```

Comment:
Is manually parsing the PyDateTime object necessary? I thought pyo3 comes with support for chrono `DateTime<UTC>` (see https://pyo3.rs/main/doc/pyo3/chrono/)

basically can you not replace most of this code with 
```rust
if let Ok(date_time) = time.extract::<DateTime<UTC>>() {}
```